R/ExpHazOS.R

```diff
@@ -0,0 +1,17 @@
+#' OS Hazard Function from Constant Transition Hazards
```

Comment:
question is which file to put this into - looking at the current package structure in https://github.com/insightsengineering/simIDM/tree/main/R I don't think we want one file per one function, how about `hazardFunctions.R` or into https://github.com/insightsengineering/simIDM/blob/main/R/survivalFunctions.R ?